src/core/meshcop/dataset_manager.cpp

```diff
@@ -863,21 +869,6 @@ PendingDatasetManager::PendingDatasetManager(Instance &aInstance)
 {
 }
 
-void PendingDatasetManager::Clear(void)
-{
-    DatasetManager::Clear();
-    mDelayTimer.Stop();
-}
-
-void PendingDatasetManager::ClearNetwork(void)
-{
-    Dataset dataset;
-
-    mNetworkTimestamp.Clear();
-    mNetworkTimestampValid = false;
-    IgnoreError(DatasetManager::Save(dataset));
-}
```

Comment:
Unlike `Clear()`, the `ClearNetwork()` method does not delete the dataset in `Settings`. Instead, it saves an empty `Dataset` (with length zero). This behavior should be equivalent to deleting the `Dataset` (not clear to me if saving empty dataset was intentional).

`ClearNetwork()` is used in one place, within `Mle::HandleChildIdResponse()`, when parsing the "Child ID Response" and the response does not contain `PendingTimestampTlv`. It seems that `Clear()` can be used instead in this case.